fluentdhec/lambda_function.py

```diff
@@ -32,19 +34,41 @@ def lambda_handler(event, context):
         send_to_hec(payload)
 
 
+def extract_time(message):
+    res = False
+    regex = r'(?:[A-Z]+\s\[|usecs:)(?P<timestamp>[0-9\-\: ]+)'
```

Comment:
Instead of doing the extraction and conversion manually you might want to try https://dateparser.readthedocs.io/en/latest/ as that will try and auto detect date formats and just do the right thing. YMMV.